config/src/test/groovy/org/springframework/security/config/annotation/web/configurers/HeadersConfigurerTests.groovy

```diff
@@ -565,4 +567,20 @@ class HeadersConfigurerTests extends BaseSpringSpec {
 		    responseHeaders == ['Strict-Transport-Security': 'max-age=31536000 ; includeSubDomains ; preload']
 	}
 
+    def "headers written at beginning of request"() {
```

Comment:
Please make this a Java test. We are in the process of coverting the Groovy tests over to Java, so we don't want to make more work for ourselves.

Reply:
Done.